### PR TITLE
Give defaultClientName a value when it is null

### PR DESCRIPTION
**Pull request by jeremymeng**
This prevents calling TryGetAzureRoleInstanceIdNoThrow() repeatedly in the null
case.

### CODE REVIEW

**Comment by mgravell** [comment]
In principal: ok. But that final default value...
less OK. Something like "StackExchange.Redis" would be Ok, I guess.

**Comment by jeremymeng** [comment]
Thanks @mgravell for reviewing.  Default value updated as suggested.

**Comment by mgravell** [comment]
@NickCraver do you know anything about that temp reference that is being
removed as a side artefact here? I have zero context

**Comment by jeremymeng** [comment]
Not sure about which xunit issue either. The xunit changes fix the build (although netcore tests are not running on appveyor). In xunit 2.3.1 xunit.analyzer is pulled in implicitly.

If needed I can split xunit changes into a separate PR.

**Comment by jeremymeng** [comment]
netcoreapp tests are now running after upgrading to xunit 2.4 beta1 (and changes responding to obsoleted xunit API).

File: StackExchange.Redis.Tests/StackExchange.Redis.Tests.csproj [hunk 1]
```diff
@@ -25,8 +25,6 @@
     <PackageReference Include="Jil" Version="2.15.4" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
     <PackageReference Include="Moq" Version="4.7.99" />
-    <!-- Temp workaround for analyzer issue -->
-    <PackageReference Include="xunit.analyzers" Version="0.6.1" />
```
Comment (jeremymeng) [inline StackExchange.Redis.Tests/StackExchange.Redis.Tests.csproj]: @NickCraver do you remember which issue this workaround refers to?

**Comment by NickCraver** [comment]
Fixed conflicts here (including ref removal), that was a temp workaround for a null ref exception in a pre-release analyzer. After build here I'll get this merged in so a rebase of the netstandard2.0 bits is good to go.

**Comment by NickCraver** [comment]
Test failures are existing (which we need to work on) - this doesn't regress any. Merging in, and thanks!

**Pull request merged**